crates/shaders/src/types.rs

```diff
@@ -20,6 +20,10 @@ pub enum BindType {
 }
 
 impl BindType {
+    // TODO: This is a public method, which means it definitely is not
+    // "dead code". However, rustc seems insitent that it is, and so to not
+    // block forward progress, I shall humour it
+    #[allow(dead_code)]
```

Comment:
See https://xi.zulipchat.com/#narrow/stream/255911-rust/topic/Confusing.20method.20is.20never.20used.20warning

I can't prioritise debugging and resolving this, so think we should land this without that

Reply:
Yeah I honestly think it's a bug in Rust, the error warning doesn't make sense to me as the type `BindType` is clearly public...